Reely/Controller/FilmViewController.swift

```diff
@@ -0,0 +1,248 @@
+//
+//  FilmViewController.swift
+//  Reely
+//
+//  Created by Mimi Shahzad on 8/22/20.
+//  Copyright © 2020 Mimi Shahzad. All rights reserved.
+//
+
+import UIKit
+
+@available(iOS 13.0, *)
+class FilmViewController: UIViewController, UISearchBarDelegate, UITableViewDelegate, UITableViewDataSource{
+    // MARK: - vars and misc.
+    
+    @IBOutlet weak var searchView: UIView!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    var filmArray = [Film]()
+    var page = 1
+    var search = ""
+    var searchArray = [String]()
+    let defaults = UserDefaults.standard
+    var suggestionsController : FilmSearchController!
+    var resultsSearchController: UISearchController!
+    @IBOutlet weak var filmTableView: UITableView!
+    
+    /* custom font  */
+    let GillSans = UIFont(name: "Gill Sans", size: UIFont.labelFontSize)
+    
+    /* custom tint color  */
+    let plum = hexStringToUIColor(hex: "#554660")
+    
+    
+    // MARK: - The Child View Controller
+    private lazy var filmSearchController: FilmSearchController = {
+        // Load Storyboard
+        let storyboard = UIStoryboard(name: "Main", bundle: Bundle.main)
+        
+        // Instantiate View Controller
+        var viewController = storyboard.instantiateViewController(withIdentifier: "filmSearchController") as! FilmSearchController
+        
+        // Add View Controller as Child View Controller
+        self.addChild(viewController)
+        viewController.filmViewController = self
+        
+        return viewController
+    }()
+    
+    
+    // MARK: - viewDidLoad()
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+       /* remove history for testing
+         previously the app would crash if there was no history*/
+//        UserDefaults.standard.removeObject(forKey: "searches")
+        
+        
+        
+        /* setup the view's controller hierarchy  */ 
+        setupView()
+        
+        
+        /* make sure that the searchArray has 10 slots only  */
+        searchArray.reserveCapacity(10)
```

Comment:
see other magic number comment for this as well. It could even be a constant used through the app so you could change it once place and all search logic picks it up